src/Linode/Provider.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace SocialiteProviders\Linode;
+
+use GuzzleHttp\RequestOptions;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    public const IDENTIFIER = 'LINODE';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase(
+            'https://login.linode.com/oauth/authorize',
+            $state
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return 'https://login.linode.com/oauth/token';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get(
+            'https://api.linode.com/v4/profile',
+            [
+                RequestOptions::HEADERS => [
+                    'Authorization' => 'Bearer '.$token,
+                ],
+            ]
+        );
+
+        return json_decode((string) $response->getBody(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'id'     => $user['uid'],
+            'name'   => $user['username'],
+            'email'  => $user['email'],
+            'avatar' => 'https://www.gravatar.com/avatar/'.md5($user['email']),
```

Comment:
do we need to return this? I think it's prob best to just return null and someone can make a gravatar url if they want?

Reply:
I’m fine with omitting it. It was copied from the DigitalOcean configuration.

Reply:
Thanks :)